### PR TITLE
Live 75 tests for 403

### PR DESCRIPTION
**Pull request by pkrucz00**
Tests for given scenario: The user is trying to modify a resource that does not belong to him (for example a webinar of which he is not a moderator)

### CODE REVIEW

File: test/membrane_live_web/controllers/user_controller_test.exs [hunk 1]
```diff
@@ -20,24 +18,38 @@ defmodule MembraneLiveWeb.UserControllerTest do
...
-    {:ok, conn: conn}
+    {:ok, conn: conn, user: user}
   end
 
   describe "index" do
-    test "lists all users", %{conn: conn} do
+    test "lists all users", %{conn: conn, user: %{uuid: user_uuid}} do
       conn = get(conn, Routes.user_path(conn, :index))
-      assert json_response(conn, 200)["data"] == []
+      assert [%{"uuid" => ^user_uuid}] = json_response(conn, 200)["data"]
```
Comment (Rados13) [inline test/membrane_live_web/controllers/user_controller_test.exs:32]: Assert expects boolean as argument you instead are making assignment here.

File: test/membrane_live_web/controllers/user_controller_test.exs [hunk 2]
```diff
@@ -20,24 +18,38 @@ defmodule MembraneLiveWeb.UserControllerTest do
...
-  describe "update user" do
-    setup [:create_user]
+  describe "show" do
+    test "req is valid", %{conn: conn, user: %{uuid: uuid, email: email}} do
+      conn = get(conn, Routes.user_path(conn, :show, uuid))
+
+      assert %{
+               "uuid" => ^uuid,
+               "email" => ^email
+             } = json_response(conn, 200)["data"]
```
Comment (Rados13) [inline test/membrane_live_web/controllers/user_controller_test.exs:43]: Same as above.

File: test/membrane_live_web/controllers/user_controller_test.exs [hunk 3]
```diff
@@ -20,24 +18,38 @@ defmodule MembraneLiveWeb.UserControllerTest do
...
+    end
 
+    test "different user in header", %{conn: conn, user: user} = context do
+      callback = &get(&1, Routes.user_path(conn, :show, user))
+      test_unauthorized_user_request(callback, context)
```
Comment (Rados13) [inline test/membrane_live_web/controllers/user_controller_test.exs:48]: What do you think about this?
```suggestion
    user_path = Routes.user_path(conn, :show, user)
    callback = &get(&1, user_path)
    test_unauthorized_user_request(callback, context)
```

File: test/membrane_live_web/controllers/webinar_controller_test.exs [hunk 4]
```diff
@@ -79,6 +74,30 @@ defmodule MembraneLiveWeb.WebinarControllerTest do
...
+               "uuid" => ^webinar_uuid,
+               "description" => "some description",
+               "presenters" => [],
+               "start_date" => "2022-07-17T10:20:00",
+               "title" => "some title"
+             } = json_response(conn, 200)["webinar"]
+    end
+
+    test "reject if user is not authorized", %{conn: conn, webinar: webinar} = context do
+      callback = &get(&1, Routes.webinar_path(conn, :show, webinar))
```
Comment (Rados13) [inline test/membrane_live_web/controllers/webinar_controller_test.exs:96]: I would move `Routes.webinar_path(conn, :show, webinar)` to variable or use `then` and pipe it.

File: test/membrane_live_web/controllers/webinar_controller_test.exs [hunk 5]
```diff
@@ -128,7 +157,23 @@ defmodule MembraneLiveWeb.WebinarControllerTest do
...
+
+    conn
+    |> put_user_in_auth_header(fake_user)
+    |> then(conn_callback)
+    |> unauthorize_assert(expected_msg)
   end
+
+  def unauthorize_assert(conn, expected_msg) do
+    assert %{"message" => ^expected_msg} = json_response(conn, 403)
+  end
```
Comment (Rados13) [inline test/membrane_live_web/controllers/webinar_controller_test.exs:175]: Could be moved to utils.

File: test/membrane_live_web/controllers/user_controller_test.exs [hunk 6]
```diff
@@ -56,28 +68,45 @@ defmodule MembraneLiveWeb.UserControllerTest do
...
-  defp get_valid_bearer() do
-    with {:ok, valid_token, _claims} <- Tokens.auth_encode(@dummy_uuid) do
-      "Bearer #{valid_token}"
-    end
+  def unauthorize_assert(conn, expected_msg) do
+    assert %{"message" => ^expected_msg} = json_response(conn, 403)
   end
+
+  defp unauthorized_error_message(fake_id, user_id),
+    do: "User with uuid #{fake_id} does not have access to user with uuid #{user_id}"
```
Comment (Rados13) [inline test/membrane_live_web/controllers/user_controller_test.exs:111]: It looks very similiar to the same function with same name in `webinar_controller_test.exs`.

File: test/membrane_live_web/plug/auth_plug_integration_test.exs [hunk 7]
```diff
@@ -67,36 +72,38 @@ defmodule MembraneLiveWeb.AuthPlugIntegrationTest do
...
+  defp compute_req_args(routing_function, conn, %{endpoint_atom: atom, body: body}) do
+    [conn, compute_routing_function(routing_function, conn, atom), body]
   end
 
-  defp test_endpoint(routing_function, conn, %{endpoint_atom: :show}) do
-    conn
-    |> get(routing_function.(conn, :show, @dummy_uuid))
-    |> assert_status_is_401()
+  defp compute_req_args(routing_function, conn, %{endpoint_atom: atom}) do
+    [conn, compute_routing_function(routing_function, conn, atom)]
```
Comment (Rados13) [inline test/membrane_live_web/plug/auth_plug_integration_test.exs:90]: The same call for `compute_routing_function(routing_function, conn, atom)` in both function.

File: test/membrane_live_web/plug/auth_plug_integration_test.exs [hunk 8]
```diff
@@ -67,36 +72,38 @@ defmodule MembraneLiveWeb.AuthPlugIntegrationTest do
...
 
-  defp test_endpoint(routing_function, conn, %{endpoint_atom: :index}) do
-    conn
-    |> get(routing_function.(conn, :index))
-    |> assert_status_is_401()
+  defp compute_routing_function(routing_function, conn, atom) do
+    case :erlang.fun_info(routing_function)[:arity] do
+      2 -> routing_function.(conn, atom)
+      3 -> routing_function.(conn, atom, @dummy_uuid)
+    end
```
Comment (Rados13) [inline test/membrane_live_web/plug/auth_plug_integration_test.exs:97]: It could be called only once in `test_endpoint` function.

**Review by Rados13 — CHANGES_REQUESTED** [review]

File: lib/membrane_live/accounts.ex [hunk 9]
```diff
@@ -12,7 +12,12 @@ defmodule MembraneLive.Accounts do
     Repo.all(User)
   end
 
-  def get_user(uuid), do: Repo.get(User, uuid)
+  def get_user(uuid) do
+    case Repo.get(User, uuid) do
+      nil -> {:error, :no_user}
+      user -> {:ok, user}
+    end
+  end
```
[the remark this answers is not in the document]
Comment (Rados13) [reply lib/membrane_live/accounts.ex:20]: Where is the spec?

File: test/membrane_live_web/controllers/webinar_controller_test.exs (hunk 4 above)
Comment (Rados13) [reply test/membrane_live_web/controllers/webinar_controller_test.exs:96]: What about this?

File: test/membrane_live_web/controllers/webinar_controller_test.exs [hunk 10]
```diff
@@ -104,6 +123,11 @@ defmodule MembraneLiveWeb.WebinarControllerTest do
       conn = put(conn, Routes.webinar_path(conn, :update, webinar), webinar: @invalid_attrs)
       assert json_response(conn, 422)["errors"] != %{}
     end
+
+    test "rejects when user not authorized", %{conn: conn, webinar: webinar} = context do
+      callback = &put(&1, Routes.webinar_path(conn, :update, webinar), webinar: @update_attrs)
```
Comment (Rados13) [inline test/membrane_live_web/controllers/webinar_controller_test.exs:128]: As above.

File: .github/workflows/sandbox_build_and_deploy.yml [hunk 11]
```diff
@@ -59,7 +59,7 @@ jobs:
         uses: TapTap21/docker-remote-deployment-action@v1.1
         with:
           remote_docker_host: ${{ secrets.SANDBOX_HOST }}
-          ssh_private_key: ${{ secrets.SSH_PRIV_KEY_GA }}
```
Comment (Rados13) [inline .github/workflows/sandbox_build_and_deploy.yml:62]: Are you sure about this change?

**Review by Rados13 — APPROVED** [review]

File: .github/workflows/sandbox_build_and_deploy.yml (hunk 11 above)
Comment (pkrucz00) [reply .github/workflows/sandbox_build_and_deploy.yml:62]: It's probably a bug after merging with he feature branch. The current version is the same as on the master branch.

**Pull request merged**